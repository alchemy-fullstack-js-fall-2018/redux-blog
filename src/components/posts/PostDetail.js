import React, { Fragment, PureComponent } from 'react';
import PropTypes from 'prop-types';
import Comments from '../../containers/comments/Comments';

export default class PostDetail extends PureComponent {

  static propTypes = {
    post: PropTypes.object.isRequired,
    fetchPost: PropTypes.func.isRequired
  };

  componentDidMount() {
    this.props.fetchPost(this.props.match.params.id);
  }

  render() {
    const id = parseInt(this.props.match.params.id);
    const { title, body } = this.props.post;

    return (
      <Fragment>
        <h2>Post Detail</h2>
        <h2>Title: {title}</h2>
        <p>Body: {body}</p>
        <Comments postId={id} />
      </Fragment>
    );
  }
}
