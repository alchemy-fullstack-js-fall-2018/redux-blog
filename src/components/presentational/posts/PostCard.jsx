import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../../routes';
import styled from 'styled-components';

const StyledDiv = styled.div`
  margin: 5px;
  border: 1px solid black;
`;

const PostCard = ({ post }) => {
  const { id, userId, title, body } = post;

  return (
    <StyledDiv>
      <p>id: {id}</p>
      <p>userId: {userId}</p>
      <p>title: {title}</p>
      <p>preview: {body.slice(0, 25)}</p>
      <Link to={ROUTES.POST_DETAIL.linkTo(id)}>
        See more...
      </Link>
    </StyledDiv>
  );
};

PostCard.propTypes = {
  post: PropTypes.object.isRequired
};

export default PostCard;
