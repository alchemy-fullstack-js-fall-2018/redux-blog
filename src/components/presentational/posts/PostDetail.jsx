import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import CommentList from '../comments/CommentList.jsx';

export default class PostDetail extends PureComponent {

  static propTypes = {
    post: PropTypes.object.isRequired,
    fetchComments: PropTypes.func.isRequired
  };

  componentDidMount() {
    const { fetchComments } = this.props;
    const { id } = this.props.match.params;
    fetchComments(id);
  }

  render() {
    const { id, userId, title, body } = this.props.post;
    const { comments } = this.props;

    return (
      <div>
        <p>id: {id}</p>
        <p>userId: {userId}</p>
        <p>title: {title}</p>
        <p>body: {body}</p>
        <CommentList comments={comments} />
      </div>
    );

  }
}
