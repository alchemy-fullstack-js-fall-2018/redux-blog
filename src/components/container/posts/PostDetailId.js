import { connect } from 'react-redux';
import PostDetail from '../../presentational/posts/PostDetail.jsx';
import { getPost } from '../../../data/state/selectors/posts';

const mapStateToProps = (state, props) => ({
  post: getPost(state, props.match.params.id)
});

export default connect(
  mapStateToProps
)(PostDetail);
