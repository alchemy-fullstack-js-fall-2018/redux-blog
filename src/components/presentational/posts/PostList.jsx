import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import PostCard from './PostCard.jsx';
import Loader from '../../lib/Loader.jsx';

const StyledDiv = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
`;

export default class PostList extends PureComponent {

  static propTypes = {
    posts: PropTypes.array.isRequired,
    fetchPosts: PropTypes.func,
    postUpdateQuery: PropTypes.func,
    isLoading: PropTypes.bool
  };

  componentDidMount() {
    const { fetchPosts } = this.props;
    if(fetchPosts) this.props.fetchPosts();
  }

  render() {
    const { posts, isLoading } = this.props;
    const postList = posts.map(post => {
      return (
        <PostCard
          key={post.id}
          post={post}
        />
      );
    });

    const gif = 'https://media.giphy.com/media/l41lFw057lAJQMwg0/giphy.gif'

    return (
      <div>
        {isLoading ?
          <Loader src={gif} />
          :
          <StyledDiv>
            {postList}
          </StyledDiv>
        }
      </div>
    );

  }
}
