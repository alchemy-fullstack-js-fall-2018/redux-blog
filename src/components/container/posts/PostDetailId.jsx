import { connect } from 'react-redux';
import PostDetail from '../../presentational/posts/PostDetail.jsx';
import { getPost } from '../../../data/state/selectors/posts';
import { getComments } from '../../../data/state/selectors/comments';
import { fetchComments } from '../../../data/state/actions/comments';

const mapStateToProps = (state, props) => ({
  post: getPost(state, parseInt(props.match.params.id)),
  comments: getComments(state)
});

const mapDispatchToProps = dispatch => ({
  fetchComments: id => dispatch(fetchComments(id))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PostDetail);
