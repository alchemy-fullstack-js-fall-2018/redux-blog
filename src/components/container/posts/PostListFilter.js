import { connect } from 'react-redux';
import PostList from '../../presentational/posts/PostList.jsx';
import { getFilteredPosts, loadingPostsCheck } from '../../../data/state/selectors/posts';
import { fetchPosts, postsUpdateQuery } from '../../../data/state/actions/posts';

const mapStateToProps = state => ({
  posts: getFilteredPosts(state),
  isLoading: loadingPostsCheck(state)
});

const mapDispatchToProps = dispatch => ({
  postsUpdateQuery: query => dispatch(postsUpdateQuery(query)),
  fetchPosts: () => dispatch(fetchPosts())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PostList);
