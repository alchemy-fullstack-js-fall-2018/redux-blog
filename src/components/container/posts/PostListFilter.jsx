import { connect } from 'react-redux';
import PostList from '../../presentational/posts/PostList.jsx';
import { getFilteredPosts, loadingPostsCheck, getQuery } from '../../../data/state/selectors/posts';
import { fetchPosts, postsUpdateQuery } from '../../../data/state/actions/posts';

import React, { Fragment } from 'react';
import styled from 'styled-components';


const StyledLabel = styled.label`
  margin: 0px auto;
  display: inline;
  justify-self: end;
`;

const StyledInput = styled.input`
  justify-self: start;
  margin: 5px;
  width: 25%;
  display: inline-block;
  margin: 0px auto;
`;


const PostListFilter = ({
  posts, isLoading, fetchPosts, query, postsUpdateQuery
}) => {
  return (
    <Fragment>
      <StyledLabel htmlFor="query">Search:</StyledLabel>
      <StyledInput
        name="query" type="text"
        value={query} onChange={postsUpdateQuery}
      ></StyledInput>

      <PostList
        posts={posts}
        isLoading={isLoading}
        fetchPosts={fetchPosts}
      />
    </Fragment>
  );
};


const mapStateToProps = state => ({
  posts: getFilteredPosts(state),
  isLoading: loadingPostsCheck(state),
  query: getQuery(state)
});

const mapDispatchToProps = dispatch => ({
  postsUpdateQuery: ({ target }) => dispatch(postsUpdateQuery(target.value)),
  fetchPosts: () => dispatch(fetchPosts())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PostListFilter);
