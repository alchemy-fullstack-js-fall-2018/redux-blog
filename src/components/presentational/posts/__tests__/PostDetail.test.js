import React from 'react';
import { shallow } from 'enzyme';
import PostDetail from '../PostDetail.jsx';
import posts from '../../../../testing/fixtures/posts.json';

describe('<PostDetail />', () => {
  test('renders', () => {
    const post = posts[0];
    const wrapper = shallow(
      <PostDetail
        post={post}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
