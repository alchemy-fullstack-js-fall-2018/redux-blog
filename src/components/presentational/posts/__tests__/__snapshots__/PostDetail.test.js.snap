// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostDetail /> renders 1`] = `
<div>
  <p>
    id: 
    1
  </p>
  <p>
    userId: 
    1
  </p>
  <p>
    title: 
    sunt aut facere repellat provident occaecati excepturi optio reprehenderit
  </p>
  <p>
    body: 
    quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto
  </p>
</div>
`;
